extensions/standard-processors/processors/ReplaceText.h

```diff
@@ -0,0 +1,102 @@
+  (LAST_LINE, "Last-Line"),
+  (EXCEPT_FIRST_LINE, "Except-First-Line"),
+  (EXCEPT_LAST_LINE, "Except-Last-Line")
+)
+
+SMART_ENUM(ReplacementStrategyType,
+  (PREPEND, "Prepend"),
+  (APPEND, "Append"),
+  (REGEX_REPLACE, "Regex Replace"),
+  (LITERAL_REPLACE, "Literal Replace"),
+  (ALWAYS_REPLACE, "Always Replace"),
+  (SUBSTITUTE_VARIABLES, "Substitute Variables")
+)
+
+class ReplaceText : public core::Processor {
+ public:
+  static const core::Property EvaluationMode;
+  static const core::Property LineByLineEvaluationMode;
+  static const core::Property ReplacementStrategy;
+  static const core::Property MaximumBufferSize;
+  static const core::Property SearchValue;
+  static const core::Property ReplacementValue;
+
+  static const core::Relationship Success;
+  static const core::Relationship Failure;
+
+  explicit ReplaceText(const std::string& name, const utils::Identifier& uuid = {});
+  core::annotation::Input getInputRequirement() const override;
+  void initialize() override;
+  void onSchedule(const std::shared_ptr<core::ProcessContext>& context, const std::shared_ptr<core::ProcessSessionFactory>&) override;
+  void onTrigger(const std::shared_ptr<core::ProcessContext>& context, const std::shared_ptr<core::ProcessSession>& session) override;
+
+ private:
+  friend struct ReplaceTextTestAccessor;
+
+  void readSearchValueProperty(const std::shared_ptr<core::ProcessContext>& context, const std::shared_ptr<core::FlowFile>& flow_file);
+  void readReplacementValueProperty(const std::shared_ptr<core::ProcessContext>& context, const std::shared_ptr<core::FlowFile>& flow_file);
+
+  void replaceTextInEntireFile(const std::shared_ptr<core::FlowFile>& flow_file, const std::shared_ptr<core::ProcessSession>& session) const;
+  void replaceTextLineByLine(const std::shared_ptr<core::FlowFile>& flow_file, const std::shared_ptr<core::ProcessSession>& session) const;
+
+  std::string applyReplacements(const std::string& input, const std::shared_ptr<core::FlowFile>& flow_file) const;
+  std::string applyRegexReplace(const std::string& input) const;
+  std::string applyLiteralReplace(const std::string& input) const;
+  std::string applySubstituteVariables(const std::string& input, const std::shared_ptr<core::FlowFile>& flow_file) const;
+  std::string getAttributeValue(const std::shared_ptr<core::FlowFile>& flow_file, const std::smatch& match) const;
+
+  std::shared_ptr<logging::Logger> logger_;
+  std::string search_value_;
+  std::regex search_regex_;
+  std::string replacement_value_;
+  uint64_t maximum_buffer_size_ = 0;
+  EvaluationModeType evaluation_mode_ = EvaluationModeType::LINE_BY_LINE;
+  LineByLineEvaluationModeType line_by_line_evaluation_mode_ = LineByLineEvaluationModeType::ALL;
+  ReplacementStrategyType replacement_strategy_ = ReplacementStrategyType::REGEX_REPLACE;
+};
+
+REGISTER_RESOURCE(ReplaceText, "Updates the content of a FlowFile by replacing parts of it using various replacement strategies.");
```

Comment:
note: after the dynamic PR, resource registration should happen in cpp files

Reply:
OK, I'll move it after #1138 is merged.

Reply:
done